CHANGELOG.md

```diff
@@ -16,6 +16,7 @@ from Dashboard to Kubernetes environment. [#481](https://github.com/TykTechnolog
 - Fixed user email format used in integration tests since e2e tests were failing after Tyk v4.0.6 was released [#510](https://github.com/TykTechnologies/tyk-operator/pull/510)
 - Fixed bug in linking logic of SubGraph CR and ApiDefinition CR [#522](https://github.com/TykTechnologies/tyk-operator/pull/522).
 - Operator was panicking when invalid certificate was provided [#529](https://github.com/TykTechnologies/tyk-operator/pull/529)
+- Operator was failing to remove finalizers from ApiDefinition that was already deleted in Dashboard
```

Comment:
You need to make an entry under [Unreleased section](https://github.com/TykTechnologies/tyk-operator/blob/585124ee692701f3bd80301988be3d91508889b0/CHANGELOG.md#unreleased)